docs/v1/P4Runtime-Spec.mdk

```diff
@@ -5381,4 +5381,21 @@ entity {
 ~ End Prototext
 
 
+## Guidelines for Implementations
+
+This section contains practical advice for using P4Runtime.
+
+### gRPC Metadata Maximum Size
+
+In gRPC, the status of a RPC request is sent as metadata, whose size is limited by the `grpc.max_metadata_size` gRPC channel argument.  By default, this limit is 8KB, which can be a problem for the `Write` P4Runtime RPC.  The `Write` RPC returns an individual error for every item in a batch (see Section [#sec-write-rpc]), which can quickly result in a status size over 8KB.  In that case, the gRPC server would not send the status, but instead send a `RESOURCE_EXHAUSTED` error, without any of the individual errors.
+
+To fix this problem, one can set the `grpc.max_metadata_size` option on the client channel.  This allows the client to receive more than 8KB of metadata, based on the new limit.  Note that the gRPC server does not have to change it's limit, as only the receiving side's limit is relevant.  The exact limit that is required depends on the maximum batch size, the length of error messages inside every `p4.Error`, as well as any other metadata that is being sent over the gRPC channel.  As a rule of thumb, it might make sense to allow for at least `8192 + MAX_UPDATES_PER_WRITE * 100` bytes of metadata.
+
+For example, in C++ one can create a client channel as follows:
+~ Begin CPP
+::grpc::ChannelArguments arguments;
+arguments.SetInt(GRPC_ARG_MAX_METADATA_SIZE, 8192 + MAX_UPDATES_PER_WRITE * 100);
```

Comment:
could you include something like `const int MAX_UPDATES_PER_WRITE = 100000;` in the example, just in case someone thinks `MAX_UPDATES_PER_WRITE` is a standard gRPC or P4Runtime constant...

Reply:
Done.  I put 100 as the number, I don't think most people expect to use 10000 updates per batch.